### ISSUE TITLE
Config file mounting on the Docker image does not work as expected

### ISSUE THREAD

**Issue by makusu2** [hadolint/hadolint#652]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

As documented in https://hub.docker.com/r/hadolint/hadolint, if I mount a hadolint config file to `/root/.config/hadolint.yaml`, the config should take effect.

A fake example of expected behavior, where I have a config file that ignores a certain violated rule, and it successfully gets ignored:

```shell
$ cat my-config.yaml 
ignored:
  - DL3006
$ cat Dockerfile 
FROM busybox
CMD [ "echo", "hi" ]
$ docker run --rm -i -v "$(pwd)/my-config.yaml":/root/.config/hadolint.yaml hadolint/hadolint < Dockerfile 
$ echo $?
0
```

### Actual behavior

A mounted config file has no effect.

An example of actual behavior, where I have a config file that ignores a certain violated rule, but the rule is not ignored:

```shell
$ cat my-config.yaml 
ignored:
  - DL3006
$ cat Dockerfile 
FROM busybox
CMD [ "echo", "hi" ]
$ docker run --rm -i -v "$(pwd)/my-config.yaml":/root/.config/hadolint.yaml hadolint/hadolint < Dockerfile 
-:1 DL3006 warning: Always tag the version of an image explicitly
$ echo $?
1
```

### Steps to reproduce the behavior

Create this `my-config.yaml`:

```yaml
ignored:
  - DL3006
```

Create this `Dockerfile`:

```Dockerfile
FROM busybox
CMD [ "echo", "hi" ]
```

Running Hadolint directly works as expected:

```shell
$ cat my-config.yaml 
ignored:
  - DL3006
$ cat Dockerfile 
FROM busybox
CMD [ "echo", "hi" ]
$ hadolint Dockerfile
Dockerfile:1 DL3006 warning: Always tag the version of an image explicitly
$ echo $?
1
$ hadolint --config ./my-config.yaml Dockerfile
$ echo $?
0
```

Using the `hadolint/hadolint` Docker image with no provided config also works as expected:

```shell
$ docker run --rm -i hadolint/hadolint < Dockerfile 
-:1 DL3006 warning: Always tag the version of an image explicitly
$ echo $?
1
```

However, when I pass the config file as described in https://hub.docker.com/r/hadolint/hadolint, it seems to have no effect:

```shell
$ docker run --rm -i -v "$(pwd)/my-config.yaml":/root/.config/hadolint.yaml hadolint/hadolint < Dockerfile 
-:1 DL3006 warning: Always tag the version of an image explicitly
$ echo $?
1
```

And yes, I've checked with `set -x` that using `$(pwd)` expands to the absolute path expected, and I've also tried hardcoding the absolute path.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```shell
$ docker run --rm hadolint/hadolint hadolint --version
Haskell Dockerfile Linter v2.4.1-0-g4b95e40
```

**Output of `docker --version`:**

```shell
$ docker --version
Docker version 20.10.7, build f0df350
```

**Comment by NJAldwin** [comment]
The crux of the issue here seems to be that the path `/root/.config/hadolint.yaml` isn't considered by hadolint when it looks for a config file **for the `latest`-tagged (`scratch`-based) image**.

This works for me if I use the same files but use `latest-alpine` or `latest-debian`:

```console
$ docker run --rm -i -v "$(pwd)/my-config.yaml":/root/.config/hadolint.yaml hadolint/hadolint:latest-debian < Dockerfile
$ echo $?
0


$ docker run --rm -i -v "$(pwd)/my-config.yaml":/root/.config/hadolint.yaml hadolint/hadolint:latest-alpine < Dockerfile
$ echo $?
0
```

I was able to get it to work for the `scratch`-based image by mounting to `/.hadolint.yaml` (per the docs, `<curdir>/.hadolint.yaml` is on the search path, and the WD of the `latest` image is `/`):

```console
$ docker run --rm -i -v "$(pwd)/my-config.yaml":/.hadolint.yaml hadolint/hadolint < Dockerfile
$ echo $?
0
```

So the fix for this issue may be as straightforward as updating the docs to suggest mounting the file differently for the `scratch`-based image.

**Comment by leopck** [comment]
This should be updated on the Docker Hub documentation for Hadolint, after doing as @NJAldwin suggested, I got mine to work

**Comment by m-ildefons** [comment]
> I was able to get it to work for the scratch-based image by mounting to /.hadolint.yaml (per the docs, <curdir>/.hadolint.yaml is on the search path, and the WD of the latest image is /):

This is the way to go as it works on all three image flavors.
I've updated the docs both in this readme as well as the Dockerhub page so it should be good now.
